src/plugins/workspace/public/plugin.ts

```diff
@@ -11,59 +11,51 @@ import {
   AppMountParameters,
   AppNavLinkStatus,
 } from '../../../core/public';
-import { WORKSPACE_APP_ID, WORKSPACE_ID_IN_SESSION_STORAGE, PATHS } from '../common/constants';
-import { WORKSPACE_ID_QUERYSTRING_NAME } from '../../../core/public';
+import { WORKSPACE_APP_ID, PATHS } from '../common/constants';
 import { mountDropdownList } from './mount';
+import { getWorkspaceIdFromUrl } from '../../../core/public/utils';
 
 export class WorkspacesPlugin implements Plugin<{}, {}> {
   private core?: CoreSetup;
-  private addWorkspaceListener() {
-    this.core?.workspaces.client.currentWorkspaceId$.subscribe((newWorkspaceId) => {
-      try {
-        sessionStorage.setItem(WORKSPACE_ID_IN_SESSION_STORAGE, newWorkspaceId);
-      } catch (e) {
-        /**
-         * in incognize mode, this method may throw an error
-         * */
-      }
-    });
-  }
-  private getWorkpsaceIdFromQueryString(): string | null {
-    const searchParams = new URLSearchParams(window.location.search);
-    return searchParams.get(WORKSPACE_ID_QUERYSTRING_NAME);
+  private getWorkpsaceIdFromURL(): string | null {
+    return getWorkspaceIdFromUrl(window.location.href);
   }
-  private getWorkpsaceIdFromSessionStorage(): string {
-    try {
-      return sessionStorage.getItem(WORKSPACE_ID_IN_SESSION_STORAGE) || '';
-    } catch (e) {
-      /**
-       * in incognize mode, this method may throw an error
-       * */
-      return '';
+  private getPatchedUrl = (
+    url: string,
+    workspaceId: string,
+    options?: {
+      jumpable?: boolean;
     }
-  }
-  private clearWorkspaceIdFromSessionStorage(): void {
-    try {
-      sessionStorage.removeItem(WORKSPACE_ID_IN_SESSION_STORAGE);
-    } catch (e) {
-      /**
-       * in incognize mode, this method may throw an error
-       * */
+  ) => {
+    const newUrl = new URL(url, window.location.href);
+    /**
+     * Patch workspace id into path
+     */
+    const currentWorkspaceId = workspaceId;
+    newUrl.pathname = this.core?.http.basePath.remove(newUrl.pathname) || '';
+    if (currentWorkspaceId) {
```

Comment:
Seems we don't need `currentWorkspaceId` here. We can use `workspaceId` directly.